R/makephyloseqFromTreeSummarizedExperiment.R

```diff
@@ -109,6 +109,21 @@ setMethod("makePhyloseqFromTreeSummarizedExperiment",
 setMethod("makePhyloseqFromTreeSummarizedExperiment",
           signature = c(x = "TreeSummarizedExperiment"),
     function(x, ...){
+        
+        # If rowTree exists, checks if the rowTree match with rownames:
+        # tips labels are found from rownames
+        if( !is.null(rowTree(x)) && any(!( rowTree(x)$tip) %in% rownames(x)) ){
+            # Tries to prune the tree. If that does not work, gives an error. 
+            x <- tryCatch(
+                {addTaxonomyTree(x)},
```

Comment:
Instead of pruning, this generates a totally new tree.

I think you have to extract the tree and use `ape::keep.tips` with the information provided by `rowLinks`.

Reply:
Now it works differently, indeed tree is different with `keep.tip `compared to `addTaxonomyTree`. I also removed `tryCatch`, I don't know if that's needed anymore